src/records.jl

```diff
@@ -8,57 +8,47 @@ $(TYPEDFIELDS)
 ---
 $(METHODLIST)
 """
-struct UserFields{Names, Types}
+mutable struct UserFields{Names, Types}
     """Mapping of field names to values. Note that values must match the corresponding field type included in the `UserFields` `Type` parameter"""
-    values::Dict{Symbol, Any}
```

Comment:
I'm curious about this replacement here - is resetting the values with a new NamedTuple every time more performant than just modifying the dictionary values?

Reply:
Ah, I didn't meant to push this (yet).
But the main reason to change this was, that the current usage of Dicts is buggy, since it doesn't sort the fieldnames like named tuple which has: `(a=22, b=22) === (b=22, a=22)`. 
For Dicts the order is arbitrary, which leads to a bug where while reading a large pointcloud the type can suddenly change due to a different order of the names.
Of course there are other ways to fix this, but I think NamedTuple is a good fit, and usually LLVM is _really_ good at optimizing away replacing the named tuple. 
I actually think all types you're using to read the pointcloud should be completely immutable (isbits(type) == true), to make the reading & creation more performant.
One can always mutate arrays of immutables by mutating the memory in the array, which should be more performant in general and is likely the only form of mutation one needs?

Reply:
In other words:
```
read(io::IO, ::Type{UserFields{(:a, :b), Tuple{Int, Int}})::UserFields{(:a, :b), Tuple{Int, Int}}
```
Does error occasionally without this PR, since it may return `::UserFields{(:b, :a), Tuple{Int, Int}}

Reply:
Right gotcha, thanks for the explanation :)